pkg/api/register.go

```diff
@@ -111,6 +111,15 @@ func AddToScheme(scheme *runtime.Scheme) {
 		&unversioned.APIGroup{},
 		&unversioned.APIResourceList{},
 	)
+
+	// Types registered here will be kept unversioned when they are encoded.
+	// It's for backward compatibility with release 1.1. clients.
+	Scheme.AddToKeepUnverioned(
```

Comment:
Should this only be for the legacy API group?  Because when we introduce extension servers I would like to fix this and require anyone connecting to a non core group to support both "" and "v1"?  As written this means all new 1.2 groups will also have to support empty apiVersion on their types.